### PR TITLE
Bluetooth: BASS: add scan cb to scan delegator module.

### PR DESCRIPTION
**Pull request by niym-ot**
Add scan cb to scan delegator so that Application
can be notified  when assistant starts or stops
scanning.
Also state information of Broadcast Assistant
is removed as info is not used.

### CODE REVIEW

File: include/zephyr/bluetooth/audio/bap.h [hunk 1]
```diff
@@ -658,6 +658,15 @@ struct bt_bap_scan_delegator_cb {
 	int (*bis_sync_req)(struct bt_conn *conn,
 			    const struct bt_bap_scan_delegator_recv_state *recv_state,
 			    const uint32_t bis_sync_req[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS]);
+	/**
+	 * @brief Scanning state callback
+	 *
+	 * Callback triggered when scanning starts or stops.
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/bap.h:664]:
```suggestion
	 * @brief Broadcast Assistant scanning state callback
	 *
	 * Callback triggered when a Broadcast Assistant notifies the Scan Delegator about the assistants scanning state.
```

File: include/zephyr/bluetooth/audio/bap.h [hunk 2]
```diff
@@ -658,6 +658,15 @@ struct bt_bap_scan_delegator_cb {
...
+	 *
+	 * @param conn Pointer to the connection that initiated the scan.
+	 * @param is_scanning True if scanning started, false if scanning stopped.
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/bap.h:667]:
```suggestion
	 * @param is_scanning true if scanning started, false if scanning stopped.
```

File: subsys/bluetooth/audio/bap_scan_delegator.c [hunk 3]
```diff
@@ -82,7 +77,7 @@ struct bass_recv_state_internal {
 
 struct bt_bap_scan_delegator_inst {
 	uint8_t next_src_id;
-	struct broadcast_assistant assistant_configs[CONFIG_BT_MAX_CONN];
+	struct bt_conn *assistant_conns[CONFIG_BT_MAX_CONN];
```
Comment (Thalley) [inline subsys/bluetooth/audio/bap_scan_delegator.c:80]:
```suggestion
```

I think we can remove this entirely. It is not used for anything, is it?

File: tests/bsim/bluetooth/audio/src/bap_broadcast_sink_test.c [hunk 4]
```diff
@@ -410,7 +410,14 @@ static void broadcast_code_cb(struct bt_conn *conn,
 	memcpy(recv_state_broadcast_code, broadcast_code, BT_AUDIO_BROADCAST_CODE_SIZE);
 }
 
+static void scanning_state_cb(struct bt_conn *conn, bool is_scanning)
+{
+	printk("Assistant scanning %s\n", is_scanning ? "started" : "stopped");
+
```
Comment (Thalley) [inline tests/bsim/bluetooth/audio/src/bap_broadcast_sink_test.c:416]:
```suggestion
	printk("Assistant scanning %s\n", is_scanning ? "started" : "stopped");
```

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by Thalley — APPROVED** [review]

**Review by larsgk — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**